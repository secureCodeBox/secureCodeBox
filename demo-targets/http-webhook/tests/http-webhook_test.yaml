# SPDX-FileCopyrightText: the secureCodeBox authors
#
# SPDX-License-Identifier: Apache-2.0

suite: Full Snapshot

tests:
  - it: matches the snapshot
    chart:
      version: 0.0.0
      appVersion: 0.0.0
    set:
      labels: {foo: bar}
      annotations: {foo: bar}
      nameOverride: "foo"
      fullnameOverride: "bar"
      podSecurityContext: {fsGroup: 1234}
      securityContext:
        {
          capabilities: {drop: [ALL]},
          readOnlyRootFilesystem: true,
          runAsNonRoot: true,
          runAsUser: 1234,
        }
      resources: {requests: {cpu: "123m", memory: "123Mi"}}
      ingress:
        enabled: true
        annotations: {foo: bar}
        tls: [{secretName: chart-example-tls, hosts: [chart-example.local]}]
      nodeSelector: {foo: bar}
      tolerations:
        [{key: foo, operator: Exists, value: "value1", effect: NoSchedule}]
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - {key: foo, operator: In, values: [bar, baz]}

      serviceaccount: {create: true, annotations: {foo: bar}, name: foo}
      podAnnotations: {foo: bar}
      autoscaling.enabled: true

    asserts:
      - matchSnapshot: {}
