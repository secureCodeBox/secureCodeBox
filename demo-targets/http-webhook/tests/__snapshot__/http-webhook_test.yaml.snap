# SPDX-FileCopyrightText: the secureCodeBox authors
#
# SPDX-License-Identifier: Apache-2.0

matches the snapshot:
  1: |
    raw: |2
      1. Get the application URL by running these commands:
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        foo: bar
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: http-webhook-0.0.0
      name: bar
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: foo
      template:
        metadata:
          annotations:
            foo: bar
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: foo
        spec:
          affinity:
            nodeAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                nodeSelectorTerms:
                  - matchExpressions:
                      - key: foo
                        operator: In
                        values:
                          - bar
                          - baz
          containers:
            - image: docker.io/mendhak/http-https-echo:latest
              imagePullPolicy: IfNotPresent
              name: http-webhook
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
              resources:
                requests:
                  cpu: 123m
                  memory: 123Mi
              securityContext:
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1234
          nodeSelector:
            foo: bar
          securityContext:
            fsGroup: 1234
          serviceAccountName: bar
          tolerations:
            - effect: NoSchedule
              key: foo
              operator: Exists
              value: value1
  3: |
    apiVersion: autoscaling/v2beta1
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: http-webhook-0.0.0
      name: bar
    spec:
      maxReplicas: 100
      metrics:
        - resource:
            name: cpu
            targetAverageUtilization: 80
          type: Resource
      minReplicas: 1
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: bar
  4: |
    kind: Ingress
    metadata:
      annotations:
        foo: bar
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: http-webhook-0.0.0
      name: bar
    spec:
      rules:
        - host: chart-example.local
          http:
            paths: null
      tls:
        - hosts:
            - chart-example.local
          secretName: chart-example-tls
  5: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        foo: bar
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: http-webhook-0.0.0
      name: bar
    spec:
      ports:
        - name: http
          port: 80
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: foo
      type: ClusterIP
  6: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: http-webhook-0.0.0
      name: bar
  7: |
    apiVersion: v1
    kind: Pod
    metadata:
      annotations:
        helm.sh/hook: test-success
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: http-webhook-0.0.0
      name: bar-test-connection
    spec:
      containers:
        - args:
            - bar:80
          command:
            - wget
          image: busybox
          name: wget
      restartPolicy: Never
